### PR TITLE
Feat/board

### PR DESCRIPTION
**Pull request by nattsnb**
React base Tic Tac Toe

### CODE REVIEW

File: src/Board/index.jsx [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            makeMoveAndChangeTurn={makeMoveAndChangeTurn}
+            turn={turn}
+            gameStatus={gameStatus}
+          />
+        ))}
+      </div>
+      <div className={styles.infoBox}>
+        <h1>{h1Title}</h1>
+        <div className={styles.arrowAndPlayerContainer}>
+          <BsArrowRightShort className={arrowFirstPlayerClass} />
```
Comment (Solnick) [inline src/Board/index.jsx]:
```suggestion
          <BsArrowRightShort className={turn === Turn.playerOne ? styles.arrowForPlayerOn : null} />
```
This way you can get rid of 2 (`arrowFirstPlayerClass`, `arrowSecondPlayerClass`) state variables

File: src/Board/index.jsx [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      setArrowToSecondPlayer();
+      checkIfWonOrDrawAndSetGameStatus();
+    } else if (!field.disabled && turn === Turn.playerTwo) {
+      field.value = turn;
+      setBoard({ ...board });
+      setTurn(Turn.playerOne);
+      setArrowToFirstPlayer();
+      checkIfWonOrDrawAndSetGameStatus();
+    }
+  };
```
Comment (Solnick) [inline src/Board/index.jsx]: First of all, this function should look like this:
```suggestion
  const move = (fieldId) => {
    const field = getFieldById(fieldId);
    field.value = turn;

    if (!field.disabled && turn === Turn.playerOne) {
      setTurn(Turn.playerTwo);
      setArrowToSecondPlayer();
    } else if (!field.disabled && turn === Turn.playerTwo) {
      setTurn(Turn.playerOne);
      setArrowToFirstPlayer();
    }
     setBoard({ ...board });
     checkIfWonOrDrawAndSetGameStatus();
  };
```
But…
Thanks to simplifying the state by removing `arrowFirstPlayerClass` and `arrowSecondPlayerClass`, this function will be simplified even more. It is up to you to finish this code simplification.

File: src/Board/index.jsx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        setH1Title(H1Title.playerWon);
+        showWinner();
+      }
+    }
+  };
+
+  const checkIfWonVertically = () => {
+    for (let i = 0; i < Object.keys(board).length; i++) {
+      let horizontalResultArray = [];
+      for (const row of Object.values(board)) {
```
Comment (Solnick) [inline src/Board/index.jsx]: Please make this code more readable by introducing new variables/constans or splitting it into well-named functions.

File: src/Board/index.jsx [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  );
+
+  const checkIfWonOrDrawAndSetGameStatus = () => {
+    checkIfWonHorizontally();
+    checkIfWonVertically();
+    checkIfWonDiagonally();
+    checkIfDraw();
+  };
+
+  const checkConditionsToWin = (array) => {
```
Comment (Solnick) [inline src/Board/index.jsx]: this function name is misleading

Comment (Solnick) [reply src/Board/index.jsx]: it is more basic/generic function than the tick tack toe game win condition.

**Review by Solnick — CHANGES_REQUESTED** [review]

File: src/Board/index.jsx (hunk 1 above)
Comment (nattsnb) [reply src/Board/index.jsx]: done

File: src/Board/index.jsx (hunk 4 above)
Comment (nattsnb) [reply src/Board/index.jsx]: done

File: src/Board/index.jsx (hunk 2 above)
Comment (nattsnb) [reply src/Board/index.jsx]: done

File: src/Board/index.jsx (hunk 3 above)
Comment (nattsnb) [reply src/Board/index.jsx]: done

File: src/Board/board.module.css [hunk 5]
```diff
@@ -46,3 +46,22 @@
...
+  color: red;
+  padding-left: 40px;
+
+}
+
+.winnerMessageOff {
+  visibility: hidden;
+}
+
+.playerNamesOn{
```
Comment (Solnick) [inline src/Board/board.module.css]: get rid of this

File: src/Board/index.jsx [hunk 6]
```diff
@@ -199,14 +179,37 @@ const Board = ({ firstPlayerName, secondPlayerName }) => {
...
+                  turn === Turn.playerTwo
+                      ? styles.arrowForPlayerOn
+                      : styles.arrowForPlayerOff
+                }
+            />
+            <p className={styles.paragraphPlayerName}>{secondPlayerName}</p>
+          </div>
         </div>
+        <div className={
+          gameStatus === GameStatus.gameOver
```
Comment (Solnick) [inline src/Board/index.jsx]: Hiding and showing elements in react should be done using `&&` or ternary operator.

File: src/Board/index.jsx [hunk 7]
```diff
@@ -134,53 +137,30 @@ const Board = ({ firstPlayerName, secondPlayerName }) => {
...
-    setArrowSecondPlayerClass(styles.arrowForPlayerOff);
-    setArrowFirstPlayerClass(styles.arrowForPlayerOff);
+    setTurn(Turn.none);
   };
-  const showWinner = () => {
-    if (turn === Turn.playerOne) {
-      setArrowToFirstPlayer();
-    } else {
-      setArrowToSecondPlayer();
+  const showWinner = (winnerFieldValue) => {
```
Comment (Solnick) [inline src/Board/index.jsx:159]: fix code formatting

**Review by Solnick — CHANGES_REQUESTED** [review]

File: src/Board/index.jsx (hunk 6 above)
Comment (nattsnb) [reply src/Board/index.jsx]: done

File: src/Board/board.module.css (hunk 5 above)
Comment (nattsnb) [reply src/Board/board.module.css]: done

**Review by Solnick — APPROVED** [review]

**Pull request merged**